style_guide.md

```diff
@@ -0,0 +1,270 @@
+# Yield Style Guide
```

Comment:
So this is from the [Solidity Style Guide](https://docs.soliditylang.org/en/v0.8.14/style-guide.html), correct?

My initial thought is we just link/reference the solidity style guide and then we list out topics that we diverge.  

We can discuss this more on our call.

Also wondering if we should fork the Solidity repo and then treat that fork as a Forge dependency (submodule) of this repo.  

Also we should do this all on a Yield repo.

Reply:
I'm just going to make some random notes as I go through this, it will help me jog my memory during our call.

Reply:
I took some snippets from the Solidity Style guide - at my discretion and based on what I thought might be sensible and not overtly dogmatic. 

I like the idea of simply linking to the original latest version of the doc, and we list the topics we either diverge from and/or pick out select points (which may or may not be present in the official guide) which are crucial and not to be compromised on.

Trying to find a linear balance between minimal reading and maximum comprehension. Let's discuss further :)